payjoin/src/receive/mod.rs

```diff
@@ -506,7 +628,7 @@ impl ProvisionalProposal {
             additional_fee = max_additional_fee_contribution;
```

Comment:
I'm not entirely sure why this was added, seems like it was due to a limitation in rust-bitcoin at the time. Does that limitation still exist?

This seems like the place where receivers should be allowed to contribute additional fees.

Reply:
I think I left this logic & comment to simplify the fee application logic so I could ship something I knew was correct, compatible with existing implementations, and that didn't have a receiver accidentally overpay. It could, and perhaps must, change.